File: src/main/java/nu/peg/slack/pt/di/factory/LocationServiceFactory.java
package nu.peg.slack.pt.di.factory;


import nu.peg.slack.pt.api.transport.TransportApi;
import nu.peg.slack.pt.service.LocationService;
import nu.peg.slack.pt.service.internal.DefaultLocationService;

import org.glassfish.hk2.api.Factory;

import javax.inject.Inject;

public class LocationServiceFactory implements Factory<LocationService> {

    private TransportApi transportApi;

    @Inject
    public LocationServiceFactory(TransportApi transportApi) {
        this.transportApi = transportApi;
    }

    @Override
    public LocationService provide() {
        return new DefaultLocationService(transportApi);
    }

    @Override
    public void dispose(LocationService instance) {

    }
}

File: src/main/kotlin/nu/peg/slack/pt/di/factory/LocationServiceFactory.kt
package nu.peg.slack.pt.di.factory


import nu.peg.slack.pt.api.transport.TransportApi
import nu.peg.slack.pt.service.LocationService
import nu.peg.slack.pt.service.internal.DefaultLocationService

import org.glassfish.hk2.api.Factory

import javax.inject.Inject

class LocationServiceFactory @Inject
constructor(private val transportApi: TransportApi) : Factory<LocationService> {

    fun provide(): LocationService {
        return DefaultLocationService(transportApi)
    }

    fun dispose(instance: LocationService) {

    }
}
